### DISCUSSION TITLE
Testvector creation documentation

### DISCUSSION THREAD

**Discussion by bulme-sm** [hneemann/Digital#673] [Q&A] [answered]
I found Digital a few weeks ago and it seems to be a great tool for teaching.

Now I would like to know how to create more complex testvectors. 
It seems that there is more possible than I can find in the help (documentation). 

Any helpful tipps where I can find more info on which expressions, programming language can be used would be very very welcome. E.g. how do I make an exclusive or? Which math and bit functions are allowed? Or can I use random numbers, which expressions can I use to calculate the results....

Possibly somebody has links which are helpful or can tell me which programming language is behind.

Please forgive me in case this is documented and I didn't find it.

**Comment by hneemann** [comment, marked as answer]
The test data input dialog has a help button, which displays a help text. And there is described almost everything that is possible. 

With the expressions almost everything is possible, which is also possible in Java. Exclusive or is expressed e.g. by a '^'.

In addition, almost all examples contain test cases. There you can also look around.

**Reply by bulme-sm** [reply]
Thanks, seems to be clearer now!
